mesonbuild/options.py

```diff
@@ -473,8 +474,66 @@ def add_to_argparse(self, name: str, parser: argparse.ArgumentParser, help_suffi
     OptionKey('purelibdir', module='python'): {},
 }
 
-
 class OptionStore:
```

Comment:
Why not derive this class directly from `dict` ? That would simplify it a little bit, and remove one indirection.

Reply:
The entire point of the class is that it has _only_ the methods explicitly written. That is, it does _not_ have an indexing operation so users are forced to use the semantically meaningful named methods. At the moment they are simple direct accesses but in the future (actually already in the present in the optionrefactor branch) they do more like do per-subproject overrides transparently in one centralised place.

This also allows you to add breakpoints to the accessor methods which is exremely useful.